### PR TITLE
Add the ability to the UI to delete engagements from the engagement tab

### PR DESCRIPTION
**Pull request by coheigea**
The products and findings tabs have three vertical dots that allow you to edit/delete/etc., however these are missing from the engagements tab. So if you go to "All Engagements" for example you can't delete an engagement.

This PR adds the three vertical dots for "All Engagements", "Active Engagements" and "Engagements by Product"

### CODE REVIEW

**Review by StefanFl — CHANGES_REQUESTED** [review]
The menus behave differently from the others in DefectDojo. When there are only a few datasets, the menu is within the list and not completely visible:
2022-07-13 08_14_15-Active Engagements _ DefectDojo

It should be like for example the products:
2022-07-13 08_14_35-Product List _ DefectDojo

**Comment by coheigea** [comment]
Yeah I tried and failed to find a way to display it outside of the list

**Comment by devGregA** [comment]
@blakeaowens is going to try to fix the zindex issue.

File: dojo/templates/dojo/engagement.html [hunk 1]
```diff
@@ -62,6 +64,81 @@ <h3 class="has-filters">
 
                         {% for e in engagements %}
                             <tr>
+                                <td class="nowrap">
+                                    <div class="align-top">
+                                        <div class="dropdown">
```
Comment (blakeaowens) [inline dojo/templates/dojo/engagement.html:69]:
```suggestion
                                        <div class="dropdown" style="position: fixed!important;">
```

File: dojo/templates/dojo/engagement.html [hunk 2]
```diff
@@ -62,6 +64,81 @@ <h3 class="has-filters">
...
+                                            <a href="#" class="dropdown-toggle pull-left" data-toggle="dropdown">&nbsp;<i class="fa fa-ellipsis-v"></i>&nbsp;</a>
+                                            <ul class="dropdown-menu">
```
Comment (blakeaowens) [inline dojo/templates/dojo/engagement.html:71]:
```suggestion
                                            <ul class="dropdown-menu" style="position: unset!important;">
```

File: dojo/templates/dojo/engagements_all.html [hunk 3]
```diff
@@ -47,7 +48,81 @@ <h3 class="has-filters">
                         {% for p in products %}
                             {% for e in p.engagement_set.all %}
                                 <tr>
-
+                                    <td class="nowrap">
+                                        <div class="align-top">
+                                            <div class="dropdown">
```
Comment (blakeaowens) [inline dojo/templates/dojo/engagements_all.html:53]:
```suggestion
                                            <div class="dropdown" style="position: fixed!important;">
```

File: dojo/templates/dojo/engagements_all.html [hunk 4]
```diff
@@ -47,7 +48,81 @@ <h3 class="has-filters">
...
+                                                <a href="#" class="dropdown-toggle pull-left" data-toggle="dropdown">&nbsp;<i class="fa fa-ellipsis-v"></i>&nbsp;</a>
+                                                <ul class="dropdown-menu">
```
Comment (blakeaowens) [inline dojo/templates/dojo/engagements_all.html:55]:
```suggestion
                                                <ul class="dropdown-menu"  style="position: unset!important;">
```

**Review by blakeaowens — CHANGES_REQUESTED** [review]
Screen Shot 2022-10-25 at 9 31 02 PM

While it is not identical to other places in DefectDojo, this seems to do the trick.

**Comment by coheigea** [comment]
There's still a bug in active engagements, this time if you scroll down the page the 3 dots remain in situ:

Screenshot 2022-10-26 at 12 13 15

**Comment by damiencarol** [comment]
@coheigea if this PR is still in progress, you can mark it as a draft.

**Comment by blakeaowens** [comment]
> There's still a bug in active engagements, this time if you scroll down the page the 3 dots remain in situ:
> 
> Screenshot 2022-10-26 at 12 13 15

**Comment by blakeaowens** [comment]
@coheigea I will have another look later this evening.

File: dojo/templates/dojo/engagement.html [hunk 5]
```diff
@@ -62,6 +64,81 @@ <h3 class="has-filters">
 
                         {% for e in engagements %}
                             <tr>
+                                <td class="nowrap">
+                                    <div class="align-top">
+                                        <div class="dropdown" style="position: fixed!important;">
```
Comment (blakeaowens) [inline dojo/templates/dojo/engagement.html:69]:
```suggestion
                                        <div class="dropdown">
```

File: dojo/templates/dojo/engagement.html [hunk 6]
```diff
@@ -62,6 +64,81 @@ <h3 class="has-filters">
...
+                                            <a href="#" class="dropdown-toggle pull-left" data-toggle="dropdown">&nbsp;<i class="fa fa-ellipsis-v"></i>&nbsp;</a>
+                                            <ul class="dropdown-menu" style="position: unset!important;">
```
Comment (blakeaowens) [inline dojo/templates/dojo/engagement.html:71]:
```suggestion
                                            <ul class="dropdown-menu">
```

**Comment by blakeaowens** [comment]
engagements-dropdowns

**Comment by coheigea** [comment]
LGTM thanks @blakeaowens

**Review by blakeaowens — APPROVED** [review]

**Pull request merged**